airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/CdcMySqlSourceTest.java

```diff
@@ -209,6 +209,76 @@ private void setupForComparisonBetweenFullRefreshAndCDCSnapshot(ImmutableList<Js
     ((ObjectNode) config).put("database", "test_schema");
   }
 
+  @Test
+  public void dateTimeDataTypeTest() throws Exception {
+    JsonNode record1 = Jsons.jsonNode(ImmutableMap.of(
+        "id", 1,
+        "datetime_col", "\'2013-09-05T10:10:02\'"));
+    JsonNode record2 = Jsons.jsonNode(ImmutableMap.of(
+        "id", 2,
+        "datetime_col", "\'2013-09-06T10:10:02\'"));
+    ImmutableList<JsonNode> records = ImmutableList.of(record1, record2);
+    setupForDateTimeDataTypeTest(records);
+    Set<JsonNode> originalData = records.stream().peek(c -> {
+      String dateTimeValue = c.get("datetime_col").asText();
+      ((ObjectNode) c).put("datetime_col", dateTimeValue.substring(1, dateTimeValue.length() - 1));
+    }).collect(Collectors.toSet());
+
+    AirbyteCatalog discover = source.discover(config);
+    List<AirbyteStream> streams = discover.getStreams();
+
+    assertEquals(streams.size(), 1);
+    JsonNode jsonSchema = streams.get(0).getJsonSchema().get("properties");
+    assertEquals(jsonSchema.get("id").get("type").asText(), "number");
+    assertEquals(jsonSchema.get("datetime_col").get("type").asText(), "string");
+
+    AirbyteCatalog catalog = new AirbyteCatalog().withStreams(streams);
+    final ConfiguredAirbyteCatalog configuredCatalog = CatalogHelpers.toDefaultConfiguredCatalog(catalog);
+
+    configuredCatalog.getStreams().forEach(c -> c.setSyncMode(SyncMode.INCREMENTAL));
+    Set<JsonNode> dataFromDebeziumSnapshot =
+        extractRecordMessages(AutoCloseableIterators.toListAndClose(source.read(config, configuredCatalog, null)))
+            .stream()
+            .map(airbyteRecordMessage -> {
+              JsonNode data = airbyteRecordMessage.getData();
+              removeCDCColumns((ObjectNode) data);
+              return data;
+            })
+            .collect(Collectors.toSet());
+
+    assertEquals(originalData, dataFromDebeziumSnapshot);
+
+    // TODO: Fix full refresh (non-cdc) mode. The value of the datetime_col is adjusted by the TIMEZONE
+    // the code is running in,
+    // in my case it got adjusted to IST i.e. "2013-09-05T15:40:02Z" and "2013-09-06T15:40:02Z".
```

Comment:
Not ideal, I assume it's always been like this? Is there any warning for user or silent timezone adjustment?

Reply:
Currently no. Its silent adjustment